### ISSUE TITLE
SOLOV2训练时batchsize设置无效

### ISSUE THREAD

**Issue by legendxty** [PaddlePaddle/PaddleDetection#1861]
试了下batchsize设置到8、32，和设置成2基本没啥区别，训练时显存的占用始终是10G左右。看了一圈代码也没找到原因，烦请大佬们指导一下。我的配置文件如下：
## solov2_r101_vd_fpn_3x.yml
architecture: SOLOv2
use_gpu: true
max_iters: 27000
snapshot_iter: 1000
log_smooth_window: 20
save_dir: output
pretrain_weights: https://paddle-imagenet-models-name.bj.bcebos.com/ResNet101_vd_pretrained.tar
metric: COCO
weights: output/solov2_r101/cls/model_final
num_classes: 9
use_ema: true
ema_decay: 0.9998

SOLOv2:
  backbone: ResNet
  fpn: FPN
  bbox_head: SOLOv2Head
  mask_head: SOLOv2MaskHead

ResNet:
  depth: 101
  feature_maps: [2, 3, 4, 5]
  freeze_at: 2
  norm_type: bn
  dcn_v2_stages: [3, 4, 5]
  variant: d

FPN:
  max_level: 6
  min_level: 2
  num_chan: 256
  spatial_scale: [0.03125, 0.0625, 0.125, 0.25]
  reverse_out: True

SOLOv2Head:
  seg_feat_channels: 512
  stacked_convs: 4
  num_grids: [40, 36, 24, 16, 12]
  kernel_out_channels: 256
  solov2_loss: SOLOv2Loss
  mask_nms: MaskMatrixNMS
  dcn_v2_stages: [0, 1, 2, 3]

SOLOv2MaskHead:
  in_channels: 128
  out_channels: 256
  start_level: 0
  end_level: 3
  use_dcn_in_tower: True

SOLOv2Loss:
  ins_loss_weight: 3.0
  focal_loss_gamma: 2.0
  focal_loss_alpha: 0.25

MaskMatrixNMS:
  pre_nms_top_n: 500
  post_nms_top_n: 100

LearningRate:
  base_lr: 0.001
  schedulers:
  - !PiecewiseDecay
    gamma: 0.8
    milestones: [3000, 6000, 9000, 12000, 15000, 18000, 21000, 24000]
  - !LinearWarmup
    start_factor: 0.
    steps: 50

## solov2_reader.yml
TrainReader:
  batch_size: 8
  worker_num: 2
  inputs_def:
    fields: ['image', 'im_id', 'gt_segm']
  dataset:
    !COCODataSet
    dataset_dir: dataset/cls/
    anno_path: annotations/instance_train.json
    image_dir: train
  sample_transforms:
  - !DecodeImage
    to_rgb: true
  - !Poly2Mask {}
  - !ResizeImage
    target_size: 800
    max_size: 1333
    interp: 1
    use_cv2: true
    resize_box: true
  - !RandomFlipImage
    prob: 0.5
  - !NormalizeImage
    is_channel_first: false
    is_scale: true
    mean: [0.485,0.456,0.406]
    std: [0.229, 0.224,0.225]
  - !Permute
    to_bgr: false
    channel_first: true
  batch_transforms:
  - !PadBatch
    pad_to_stride: 32
  - !Gt2Solov2Target
    num_grids: [40, 36, 24, 16, 12]
    scale_ranges: [[1, 96], [48, 192], [96, 384], [192, 768], [384, 2048]]
    coord_sigma: 0.2
  shuffle: True

EvalReader:
  inputs_def:
    fields: ['image', 'im_info', 'im_id']
  dataset:
    !COCODataSet
    dataset_dir: dataset/cls/
    anno_path: annotations/instance_val.json
    image_dir: val
  sample_transforms:
  - !DecodeImage
    to_rgb: true
  - !ResizeImage
    interp: 1
    max_size: 1333
    target_size: 800
    use_cv2: true
  - !NormalizeImage
    is_channel_first: false
    is_scale: true
    mean: [0.485,0.456,0.406]
    std: [0.229, 0.224,0.225]
  - !Permute
    channel_first: true
    to_bgr: false
  batch_transforms:
  - !PadBatch
    pad_to_stride: 32
    use_padded_im_info: false

  batch_size: 1
  shuffle: false
  drop_last: false
  drop_empty: false
  worker_num: 2

TestReader:
  inputs_def:
    fields: ['image', 'im_info', 'im_id', 'im_shape']
  dataset:
    !ImageFolder
    anno_path: dataset/coco/annotations/instances_val2017.json
  sample_transforms:
  - !DecodeImage
    to_rgb: true
  - !ResizeImage
    interp: 1
    max_size: 1333
    target_size: 800
    use_cv2: true
  - !NormalizeImage
    is_channel_first: false
    is_scale: true
    mean: [0.485,0.456,0.406]
    std: [0.229, 0.224,0.225]
  - !Permute
    channel_first: true
    to_bgr: false
  batch_transforms:
  - !PadBatch
    pad_to_stride: 32
    use_padded_im_info: false

**Comment by legendxty** [comment]
> 试了下batchsize设置到8、32，和设置成2基本没啥区别，训练时显存的占用始终是10G左右。看了一圈代码也没找到原因，烦请大佬们指导一下。我的配置文件如下：
> 
> ## solov2_r101_vd_fpn_3x.yml
> architecture: SOLOv2
> use_gpu: true
> max_iters: 27000
> snapshot_iter: 1000
> log_smooth_window: 20
> save_dir: output
> pretrain_weights: https://paddle-imagenet-models-name.bj.bcebos.com/ResNet101_vd_pretrained.tar
> metric: COCO
> weights: output/solov2_r101/cls/model_final
> num_classes: 9
> use_ema: true
> ema_decay: 0.9998
> 
> SOLOv2:
> backbone: ResNet
> fpn: FPN
> bbox_head: SOLOv2Head
> mask_head: SOLOv2MaskHead
> 
> ResNet:
> depth: 101
> feature_maps: [2, 3, 4, 5]
> freeze_at: 2
> norm_type: bn
> dcn_v2_stages: [3, 4, 5]
> variant: d
> 
> FPN:
> max_level: 6
> min_level: 2
> num_chan: 256
> spatial_scale: [0.03125, 0.0625, 0.125, 0.25]
> reverse_out: True
> 
> SOLOv2Head:
> seg_feat_channels: 512
> stacked_convs: 4
> num_grids: [40, 36, 24, 16, 12]
> kernel_out_channels: 256
> solov2_loss: SOLOv2Loss
> mask_nms: MaskMatrixNMS
> dcn_v2_stages: [0, 1, 2, 3]
> 
> SOLOv2MaskHead:
> in_channels: 128
> out_channels: 256
> start_level: 0
> end_level: 3
> use_dcn_in_tower: True
> 
> SOLOv2Loss:
> ins_loss_weight: 3.0
> focal_loss_gamma: 2.0
> focal_loss_alpha: 0.25
> 
> MaskMatrixNMS:
> pre_nms_top_n: 500
> post_nms_top_n: 100
> 
> LearningRate:
> base_lr: 0.001
> schedulers:
> 
> * !PiecewiseDecay
>   gamma: 0.8
>   milestones: [3000, 6000, 9000, 12000, 15000, 18000, 21000, 24000]
> * !LinearWarmup
>   start_factor: 0.
>   steps: 50
> 
> ## solov2_reader.yml
> TrainReader:
> batch_size: 8
> worker_num: 2
> inputs_def:
> fields: ['image', 'im_id', 'gt_segm']
> dataset:
> !COCODataSet
> dataset_dir: dataset/cls/
> anno_path: annotations/instance_train.json
> image_dir: train
> sample_transforms:
> 
> * !DecodeImage
>   to_rgb: true
> * !Poly2Mask {}
> * !ResizeImage
>   target_size: 800
>   max_size: 1333
>   interp: 1
>   use_cv2: true
>   resize_box: true
> * !RandomFlipImage
>   prob: 0.5
> * !NormalizeImage
>   is_channel_first: false
>   is_scale: true
>   mean: [0.485,0.456,0.406]
>   std: [0.229, 0.224,0.225]
> * !Permute
>   to_bgr: false
>   channel_first: true
>   batch_transforms:
> * !PadBatch
>   pad_to_stride: 32
> * !Gt2Solov2Target
>   num_grids: [40, 36, 24, 16, 12]
>   scale_ranges: [[1, 96], [48, 192], [96, 384], [192, 768], [384, 2048]]
>   coord_sigma: 0.2
>   shuffle: True
> 
> EvalReader:
> inputs_def:
> fields: ['image', 'im_info', 'im_id']
> dataset:
> !COCODataSet
> dataset_dir: dataset/cls/
> anno_path: annotations/instance_val.json
> image_dir: val
> sample_transforms:
> 
> * !DecodeImage
>   to_rgb: true
> * !ResizeImage
>   interp: 1
>   max_size: 1333
>   target_size: 800
>   use_cv2: true
> * !NormalizeImage
>   is_channel_first: false
>   is_scale: true
>   mean: [0.485,0.456,0.406]
>   std: [0.229, 0.224,0.225]
> * !Permute
>   channel_first: true
>   to_bgr: false
>   batch_transforms:
> * !PadBatch
>   pad_to_stride: 32
>   use_padded_im_info: false
> 
> batch_size: 1
> shuffle: false
> drop_last: false
> drop_empty: false
> worker_num: 2
> 
> TestReader:
> inputs_def:
> fields: ['image', 'im_info', 'im_id', 'im_shape']
> dataset:
> !ImageFolder
> anno_path: dataset/coco/annotations/instances_val2017.json
> sample_transforms:
> 
> * !DecodeImage
>   to_rgb: true
> * !ResizeImage
>   interp: 1
>   max_size: 1333
>   target_size: 800
>   use_cv2: true
> * !NormalizeImage
>   is_channel_first: false
>   is_scale: true
>   mean: [0.485,0.456,0.406]
>   std: [0.229, 0.224,0.225]
> * !Permute
>   channel_first: true
>   to_bgr: false
>   batch_transforms:
> * !PadBatch
>   pad_to_stride: 32
>   use_padded_im_info: false

解决了，应该是在这里设置： solov2_r101_vd_fpn_3x.yml
_READER_: 'solov2_reader.yml'
TrainReader:
  batch_size: 8
